### PR TITLE
[css-text-3] Spaces don't hang when break-spaces is specified

### LINKED ISSUE #2465: [css-text-3] Audit details of break-spaces value against use cases
https://lists.w3.org/Archives/Public/www-archive/2018Mar/0002.html

### PR DESCRIPTION
**Pull request by frivoal**
Closes #2465

### CODE REVIEW

File: css-text-3/Overview.bs [hunk 1]
```diff
@@ -724,17 +724,23 @@ Characters and Letters</h3>
...
-         However, if 'overflow-wrap' is set to ''break-spaces'',
-         collapsing their advance width is not allowed,
-         as this would prevent the preserved spaces from wrapping.
-
-         Note: Hanging the white space rather than collapsing it
-         allows users to see the space when selecting or editing text.
+         but have 'white-space' set to ''pre-wrap'':
+	 <ul>
+	   <li>If the value of 'overflow-wrap' is ''overflow-wrap/normal'' or ''overflow-wrap/break-word''
+	     (i.e. it does not include ''break-spaces'')
```
Comment (fantasai) [inline css-text-3/Overview.bs]: If 'overflow-wrap' does not specify ''break-spaces''

File: css-text-3/Overview.bs [hunk 2]
```diff
@@ -724,17 +724,23 @@ Characters and Letters</h3>
...
+             the UA must either <a>hang</a> the <a>white space</a>
+             or visually collapse the character advance widths
+             of any overflowing spaces
+             such that they don't take up space in the line.
+
+             Note: Hanging the white space rather than collapsing it
+             allows users to see the space when selecting or editing text.
+	   <li>Otherwise
+	     (i.e. 'overflow-wrap' property includes ''break-spaces''),
```
Comment (fantasai) [inline css-text-3/Overview.bs]: s/includes/specifies/

File: css-text-3/Overview.bs [hunk 3]
```diff
@@ -724,17 +724,23 @@ Characters and Letters</h3>
...
+             hanging or collapsing the advance width of is not allowed.
```
Comment (fantasai) [inline css-text-3/Overview.bs]: “of is” is not grammatical.

File: css-text-3/Overview.bs [hunk 4]
```diff
@@ -724,17 +724,23 @@ Characters and Letters</h3>
...
+             The end-of-lines spaces that would overflow the line are
+             wrapped as described in the definition of the ''break-spaces'' value.
```
Comment (fantasai) [inline css-text-3/Overview.bs]: Normatively, this doesn't belong here. We've already broken lines. If spaces needed to get wrapped, they've already gotten wrapped.

File: css-text-3/Overview.bs [hunk 5]
```diff
@@ -1246,14 +1252,11 @@ Line Breaking Details</h3>
       must be broken after the last white space character that would fit the line,
```
Comment (fantasai) [inline css-text-3/Overview.bs]: Here is where you specify that the spaces must take up space on the line.

File: css-text-3/Overview.bs [hunk 6]
```diff
@@ -1246,14 +1252,11 @@ Line Breaking Details</h3>
...
 
-      <p class="note">This value does not guarantee that there will never be any overflow due to spaces.
-      For example, if 'overflow-wrap' is set to ''break-spaces''
-      and 'white-space' to ''pre-wrap'',
-      and there is a single <a>preserved</a> space at the end of the line,
-      and that space overflows,
-      the rules of ''break-spaces'' do not introduce any additional line break,
-      and that space does overflow and <a>hang</a>.
-      Also, if the line length is so short that even a single space does not fit,
+      As spaces do not hang,
```
Comment (fantasai) [inline css-text-3/Overview.bs]: Otherwise this phrase can't be a reminder of that fact.

**Comment by frivoal** [comment]
@fantasai Thanks for the review, I think it is now fixed. Can you confirm (and approve), or provide further feedback?

File: css-text-3/Overview.bs [hunk 7]
```diff
@@ -1247,12 +1244,15 @@ Line Breaking Details</h3>
...
-      or after the first white space in the sequence if none would fit.
-
-      As spaces do not hang,
+      except that:
+      * Any sequence of <a>preserved</a> white space always takes up space,
+        including at the end of the line.
+      * Any sequence of <a>preserved</a> white space
+        that would otherwise overflow the line and <a>hang</a> or be collapsed as per <a href="#white-space-phase-2">Trimming and Positioning</a>
+        must be broken after the last white space character that would fit the line,
+        or after the first white space in the sequence if none would fit.
```
Comment (fantasai) [inline css-text-3/Overview.bs]: Replace this second clause with “A line breaking opportunity exists after every preserved white space character, including between white space characters.”

**Comment by fantasai** [comment]
Approved. I'm gonna let you merge, please squash and rebase if you can.

**Pull request merged**